### PR TITLE
github: Introduce a feature graduation tracker template

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @xpivarc 
/cc @vladikr 
/cc @EdDev 

### What this PR does
As discussed at the unconference for v1.5.0 [1] we need to track existing and new FGs through the various graduation phases to GA moving forward.  This commit introduces a simple GitHub template to help folks start this process.

[1] https://unconference.kubevirt.io/D62SngbSRXO1uhTAc8Ikqg#xpivarc-Feature-lifecycleFuture-of-proposals---1-1-1-1

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_gate.md
Comment (EdDev) [inline .github/ISSUE_TEMPLATE/feature_gate.md:1]: Consider calling the file `feature_lifecycle.md`.

**Review by EdDev — APPROVED** [review]
Nice, thanks!

**Comment by lyarwood** [comment]
@xpivarc @vladikr would appreciate your thoughts on this if you have time as I'd like to start creating issues for the existing FGs sooner rather than later.

**Review by EdDev — APPROVED** [review]
Thanks!

BTW, this is partially resolving https://github.com/kubevirt/community/issues/286 .
 We still need to add a recommended section in the proposal template.

**Comment by lyarwood** [comment]
> Thanks!
> 
> BTW, this is partially resolving [kubevirt/community#286](https://github.com/kubevirt/community/issues/286) . We still need to add a recommended section in the proposal template.

@EdDev  I did post https://github.com/kubevirt/community/pull/343 for that but closed it assuming folks wanted to focus on the new enhancements template instead. I can reopen it if you still think it would be useful in the short term.

**Review by iholder101 — COMMENTED** [review]
Thanks a lot @lyarwood! Very cool!

Here are some more fields that exist in Kubernetes' tracking issue (here's an example: https://github.com/kubernetes/enhancements/issues/2400) that we can consider to adopt:

One-line enhancement description:
Primary contact (assignee):
Responsible SIGs:
Design proposal link:
Important PRs (optional):

I think these would be valuable. WDYT?

**Comment by EdDev** [comment]
> @EdDev I did post [kubevirt/community#343](https://github.com/kubevirt/community/pull/343) for that but closed it assuming folks wanted to focus on the new enhancements template instead. I can reopen it if you still think it would be useful in the short term.

Ohhm I missed that one.
I think it will be useful to have it there because then at design time, both the author and reviewers need to think about it. I mean, questions like:
- Does this feature requires release stages or can it be GA directly?
- If stages are planned, is there a known plan per versions?

If you re-open it, I will ack.

**Comment by lyarwood** [comment]
> Thanks a lot @lyarwood! Very cool!
> 
> Here are some more fields that exist in Kubernetes' tracking issue (here's an example: [kubernetes/enhancements#2400](https://github.com/kubernetes/enhancements/issues/2400)) that we can consider to adopt:
> 
> One-line enhancement description: Primary contact (assignee): Responsible SIGs: Design proposal link: Important PRs (optional):
> 
> I think these would be valuable. WDYT?

@iholder101  I've added a few of these to the latest revision but to be clear I'm not attempting to codify the new enhancements approach with this tracker. Instead this is a stop gap to hopefully let us clean up the existing FGs in v1.5.0 while the new enhancements approach is discussed and agreed on. 

IMHO a new issue tracker template will likely be created for that process in v1.6.0.

> > @EdDev I did post [kubevirt/community#343](https://github.com/kubevirt/community/pull/343) for that but closed it assuming folks wanted to focus on the new enhancements template instead. I can reopen it if you still think it would be useful in the short term.
> 
> Ohhm I missed that one. I think it will be useful to have it there because then at design time, both the author and reviewers need to think about it. I mean, questions like:
> 
> * Does this feature requires release stages or can it be GA directly?
> * If stages are planned, is there a known plan per versions?
> 
> If you re-open it, I will ack.

@EdDev ACK, I'll reopen and update that now thanks.

**Comment by lyarwood** [comment]
@xpivarc @vladikr @iholder101 any more thoughts on landing this so we can at least move some of the existing FGs forward in v1.5.0 ahead of a potentially new process for tracking stuff in v1.6.0?

**Comment by iholder101** [comment]
> @iholder101 I've added a few of these to the latest revision but to be clear I'm not attempting to codify the new enhancements approach with this tracker. Instead this is a stop gap to hopefully let us clean up the existing FGs in v1.5.0 while the new enhancements approach is discussed and agreed on.
> 
> IMHO a new issue tracker template will likely be created for that process in v1.6.0.

I think that once the new proposal approach will be merged, its new feature tracker will serve as a replacement for the tracker proposed here. Since the discussion is already ongoing [1], I'm not sure we should invest in this just to replace it with the enhancement tracker in the very near future TBH.

There's always the alternative of simply documenting the feature gates in-code:
https://github.com/kubevirt/kubevirt/blob/e2e64962d9db7dc3724f751eb6c897215b5842a2/pkg/virt-config/deprecation/feature-gates.go#L50-L56

Is there a reason you think this is necessary albeit what's written above? Am I missing something here?

[1] https://github.com/kubevirt/community/pull/362

**Comment by vladikr** [comment]
@lyarwood Thank you very much!
It looks good to me. Seems like this template mimics the Kubernetes one.
My only reservation is that from my perspective this template needs to live in the (not yet existing) kubevirt/enhancements project and not in kubevirt/kubevirt.

**Comment by iholder101** [comment]
I suggest to hold this effort, wait until [#362](https://github.com/kubevirt/community/pull/362) merges, and use what's suggested here as a VEP template.

WDYT?

**Pull request closed without merging**
